Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp

```diff
@@ -244,6 +246,14 @@ NetworkProcessProxy::NetworkProcessProxy()
 #if PLATFORM(IOS_FAMILY)
     addBackgroundStateObservers();
 #endif
+
+#if ENABLE(ADVANCED_PRIVACY_PROTECTIONS)
+    m_storageAccessPromptQuirksDataUpdateObserver = StorageAccessPromptQuirkController::shared().observeUpdates([weakThis = WeakPtr { *this }] {
+        if (RefPtr protectedThis = weakThis.get())
+            protectedThis->sendStorageAccessPromptQuirksData();
+    });
+    sendStorageAccessPromptQuirksData();
```

Comment:
`updateStorageAccessPromptQuirksData()` since it should only be used for updates?